### ISSUE TITLE
`unnecessary_null_checks` false positive w/ field access

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#3365]
See: https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket/8815708920718771281/+/u/analyze_flutter_flutter/stdout

```dart

  final String? pluginClass;

  @override
  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'name': name,
      if (pluginClass != null) 'class': pluginClass!, //LINT
      if (pluginClass != null) 'filename': _filenameForCppClass(pluginClass!), // FALSE POSITIVE
      if (dartPluginClass != null) kDartPluginClass: dartPluginClass!, //LINT
      if (ffiPlugin) kFfiPlugin: true,
      if (defaultPackage != null) kDefaultPackage: defaultPackage!, //LINT
    };
  }

  String _filenameForCppClass(String className) { ... }
```

**Comment by pq** [comment]
Maybe a consequence of https://github.com/dart-lang/linter/pull/3326

/fyi @scheglov

**Comment by scheglov** [comment]
I have not tried to run any code, but looking at the PR it seems to me that the change affects only when `if (parent is MapLiteralEntry)`, and in `_filenameForCppClass(pluginClass!)` the null check `pluginClass!` is in `ArgumentList`.

**Comment by pq** [comment]
Ah.  This is OK after all.  A misreport on my end.

Sorry for the noise!
